corehq/apps/geospatial/static/geospatial/js/geospatial_map.js

```diff
@@ -290,10 +398,124 @@ hqDefine("geospatial/js/geospatial_map", [
+                            cases.push(caseModel);
+                            const coordString = caseModel.itemData.coordinates['lng'] + " " + caseModel.itemData.coordinates['lat'];
+                            caseGroupsIndex[coordString] = {groupId: groupId, item: caseModel};
+                        }
+                    });
+                    connectUserWithCasesOnMap(user, cases);
+                    groupId += 1;
+                });
+                self.setBusy(false);
+            };
+
+            self.runCaseDisbursementAlgorithm = function (cases, users) {
+                self.setBusy(true);
+                let mapInstance = map.getMapboxInstance();
+
+                let caseData = [];
+                cases.forEach(function (c) {
+                    const layerId = getLineFeatureId(c.itemId);
+                    if (mapInstance.getLayer(layerId)) {
+                        mapInstance.removeLayer(layerId);
+                    }
+                    if (mapInstance.getSource(layerId)) {
+                        mapInstance.removeSource(layerId);
+                    }
+
+                    caseData.push({
+                        id: c.itemId,
+                        lon: c.itemData.coordinates.lng,
+                        lat: c.itemData.coordinates.lat,
+                    });
+                });
+
+                let userData = users.map(function (c) {
+                    return {
+                        id: c.itemId,
+                        lon: c.itemData.coordinates.lng,
+                        lat: c.itemData.coordinates.lat,
+                    };
+                });
+
+                $.ajax({
+                    type: 'post',
+                    url: runDisbursementUrl,
+                    dataType: 'json',
+                    data: JSON.stringify({'users': userData, "cases": caseData}),
+                    contentType: "application/json; charset=utf-8",
+                    success: function (ret) {
+                        if (ret['poll_url'] !== undefined) {
+                            self.startPoll(ret['poll_url']);
+                        } else {
+                            self.handleDisbursementResults(ret['result']);
+                        }
+                    },
+                });
+            };
+
+            self.startPoll = function (pollUrl) {
+                if (!self.isBusy()) {
+                    self.setBusy(true);
+                }
+                self.pollUrl(pollUrl);
+                self.doPoll();
+            };
+
+            self.doPoll = function () {
+                var tick = function () {
+                    $.ajax({
+                        method: 'GET',
+                        url: self.pollUrl(),
+                        success: function (data) {
+                            const result = data.result;
+                            if (!data) {
+                                setTimeout(tick, 1500);
```

Comment:
Nit: Could we change 1500 into a const? Maybe something like `DEFAULT_WAIT_TIME_MS`.